apt CI failure since April 2024

Example of failing job: https://github.com/robotology/gz-sim-yarp-plugins/actions/runs/8917647874 .

The failing step is "Generate Code Coverage report":

~~~
2024-05-02T02:17:50.0725664Z ##[group]Generating code coverage report...
2024-05-02T02:17:50.0738911Z [command]/home/runner/.local/bin/gcovr --exclude build/* --exclude install/*
2024-05-02T02:17:50.1190206Z (INFO) Reading coverage data...
2024-05-02T02:17:50.4308647Z (WARNING) Unrecognized GCOV output for /home/runner/work/gz-sim-yarp-plugins/gz-sim-yarp-plugins/plugins/controlboard/src/ControlBoardTrajectory.cpp
2024-05-02T02:17:50.4310111Z 	  branch  1 taken -8
2024-05-02T02:17:50.4310638Z 	This is indicative of a gcov output parse error.
2024-05-02T02:17:50.4311350Z 	Please report this to the gcovr developers
2024-05-02T02:17:50.4311990Z 	at <https://github.com/gcovr/gcovr/issues>.
2024-05-02T02:17:50.4312580Z (WARNING) Exception during parsing:
2024-05-02T02:17:50.4315150Z 	NegativeHits: Got negative hit value in gcov line 'branch  1 taken -8' caused by a bug in gcov tool, see https://gcc.gnu.org/bugzilla/show_bug.cgi?id=68080. Use option --gcov-ignore-parse-errors with a value of negative_hits.warn, or negative_hits.warn_once_per_file.
2024-05-02T02:17:50.4317454Z (ERROR) Exiting because of parse errors.
2024-05-02T02:17:50.4318227Z 	You can run gcovr with --gcov-ignore-parse-errors
2024-05-02T02:17:50.4318931Z 	to continue anyway.
2024-05-02T02:17:50.4322004Z (ERROR) Trouble processing '/home/runner/work/gz-sim-yarp-plugins/gz-sim-yarp-plugins/build/plugins/controlboard/CMakeFiles/gz-sim-yarp-controlboard-system.dir/src/ControlBoardTrajectory.cpp.gcda' with working directory '/home/runner/work/gz-sim-yarp-plugins/gz-sim-yarp-plugins'.
2024-05-02T02:17:50.4325710Z Stdout of gcov was >>File '/home/runner/work/gz-sim-yarp-plugins/gz-sim-yarp-plugins/plugins/controlboard/src/ControlBoardTrajectory.cpp'
2024-05-02T02:17:50.4326712Z Lines executed:29.05% of 327
2024-05-02T02:17:50.4327018Z Branches executed:24.19% of 124
2024-05-02T02:17:50.4327332Z Taken at least once:15.32% of 124
2024-05-02T02:17:50.4327651Z Calls executed:16.18% of 68
2024-05-02T02:17:50.4328200Z Creating 'ControlBoardTrajectory.cpp##42e6fb7b49e901e7e9ba95a514229451.gcov'
2024-05-02T02:17:50.4328615Z 
2024-05-02T02:17:50.4328761Z File '/usr/include/c++/11/iostream'
2024-05-02T02:17:50.4329059Z No executable lines
2024-05-02T02:17:50.4329293Z No branches
2024-05-02T02:17:50.4329500Z No calls
2024-05-02T02:17:50.4329843Z Removing 'iostream##f763a46ae8c1573f45db80ce8113e1f5.gcov'
2024-05-02T02:17:50.4330132Z 
2024-05-02T02:17:50.4330297Z File '/usr/include/gz/sdformat14/sdf/Types.hh'
2024-05-02T02:17:50.4330633Z No executable lines
2024-05-02T02:17:50.4330865Z No branches
2024-05-02T02:17:50.4331067Z No calls
2024-05-02T02:17:50.4331407Z Removing 'Types.hh##cd1a16212e4b4189b47a37b44e2e3a94.gcov'
2024-05-02T02:17:50.4331688Z 
2024-05-02T02:17:50.4331837Z File '/usr/include/c++/11/bits/std_mutex.h'
2024-05-02T02:17:50.4332171Z Lines executed:0.00% of 14
2024-05-02T02:17:50.4332436Z Branches executed:0.00% of 2
2024-05-02T02:17:50.4332721Z Taken at least once:0.00% of 2
2024-05-02T02:17:50.4333245Z Calls executed:0.00% of 6
2024-05-02T02:17:50.4333666Z Creating 'std_mutex.h##51b629ed292547c2f1cc49c064941579.gcov'
2024-05-02T02:17:50.4333967Z 
2024-05-02T02:17:50.4334092Z File '/usr/include/c++/11/chrono'
2024-05-02T02:17:50.4334383Z Lines executed:100.00% of 2
2024-05-02T02:17:50.4334642Z No branches
2024-05-02T02:17:50.4334845Z No calls
2024-05-02T02:17:50.4335170Z Creating 'chrono##540ef6ae9a138064e54edd50e2f823ce.gcov'
2024-05-02T02:17:50.4335452Z 
2024-05-02T02:17:50.4335597Z File '/usr/include/c++/11/bits/std_abs.h'
2024-05-02T02:17:50.4336075Z Lines executed:0.00% of 2
2024-05-02T02:17:50.4336324Z No branches
2024-05-02T02:17:50.4336523Z No calls
2024-05-02T02:17:50.4336863Z Creating 'std_abs.h##92f205e77fee34147b655dd4ff073cd6.gcov'
2024-05-02T02:17:50.4337156Z 
2024-05-02T02:17:50.4337392Z File '/usr/include/x86_64-linux-gnu/c++/11/bits/gthr-default.h'
2024-05-02T02:17:50.4337784Z Lines executed:0.00% of 10
2024-05-02T02:17:50.4338049Z Branches executed:0.00% of 4
2024-05-02T02:17:50.4338328Z Taken at least once:0.00% of 4
2024-05-02T02:17:50.4338602Z Calls executed:0.00% of 4
2024-05-02T02:17:50.4339016Z Creating 'gthr-default.h##f59d6128c8d19146c0974ea9ba7c40d4.gcov'
2024-05-02T02:17:50.4339324Z 
2024-05-02T02:17:50.4339425Z Lines executed:27.32% of 355
2024-05-02T02:17:50.4339686Z << End of stdout
2024-05-02T02:17:50.4339937Z Stderr of gcov was >><< End of stderr
2024-05-02T02:17:50.4340867Z Current processed gcov file was '/home/runner/work/gz-sim-yarp-plugins/gz-sim-yarp-plugins/ControlBoardTrajectory.cpp##42e6fb7b49e901e7e9ba95a514229451.gcov'.
2024-05-02T02:17:50.4341767Z Use option --verbose to get extended information.
2024-05-02T02:17:50.4342166Z Traceback (most recent call last):
2024-05-02T02:17:50.4342923Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/workers.py", line 80, in worker
2024-05-02T02:17:50.4343594Z     work(*args, **kwargs)
2024-05-02T02:17:50.4344339Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/read.py", line 422, in process_datafile
2024-05-02T02:17:50.4345057Z     done = run_gcov_and_process_files(
2024-05-02T02:17:50.4345890Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/read.py", line 709, in run_gcov_and_process_files
2024-05-02T02:17:50.4346708Z     process_gcov_data(fname, abs_filename, covdata, options)
2024-05-02T02:17:50.4347576Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/read.py", line 217, in process_gcov_data
2024-05-02T02:17:50.4348315Z     coverage, source_lines = parse_coverage(
2024-05-02T02:17:50.4349136Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/parser.py", line 379, in parse_coverage
2024-05-02T02:17:50.4349834Z     _report_lines_with_errors(
2024-05-02T02:17:50.4350648Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/parser.py", line 561, in _report_lines_with_errors
2024-05-02T02:17:50.4351496Z     raise errors[0]  # guaranteed to have at least one exception
2024-05-02T02:17:50.4352370Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/parser.py", line 335, in parse_coverage
2024-05-02T02:17:50.4353168Z     _parse_line(raw_line, ignore_parse_errors, persistent_states),
2024-05-02T02:17:50.4354088Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/parser.py", line 678, in _parse_line
2024-05-02T02:17:50.4354953Z     tag = _parse_tag_line(line, ignore_parse_errors, persistent_states)
2024-05-02T02:17:50.4355923Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/parser.py", line 799, in _parse_tag_line
2024-05-02T02:17:50.4356698Z     NegativeHits.raise_if_not_ignored(
2024-05-02T02:17:50.4357585Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/parser.py", line 245, in raise_if_not_ignored
2024-05-02T02:17:50.4358504Z     raise NegativeHits(line)
2024-05-02T02:17:50.4360073Z gcovr.formats.gcov.parser.NegativeHits: Got negative hit value in gcov line 'branch  1 taken -8' caused by a bug in gcov tool, see https://gcc.gnu.org/bugzilla/show_bug.cgi?id=68080. Use option --gcov-ignore-parse-errors with a value of negative_hits.warn, or negative_hits.warn_once_per_file.
2024-05-02T02:17:50.4361712Z Traceback (most recent call last):
2024-05-02T02:17:50.4362524Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/workers.py", line 80, in worker
2024-05-02T02:17:50.4363410Z     work(*args, **kwargs)
2024-05-02T02:17:50.4364219Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/read.py", line 422, in process_datafile
2024-05-02T02:17:50.4364990Z     done = run_gcov_and_process_files(
2024-05-02T02:17:50.4365891Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/read.py", line 709, in run_gcov_and_process_files
2024-05-02T02:17:50.4366782Z     process_gcov_data(fname, abs_filename, covdata, options)
2024-05-02T02:17:50.4367711Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/read.py", line 217, in process_gcov_data
2024-05-02T02:17:50.4368503Z     coverage, source_lines = parse_coverage(
2024-05-02T02:17:50.4369348Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/parser.py", line 379, in parse_coverage
2024-05-02T02:17:50.4370055Z     _report_lines_with_errors(
2024-05-02T02:17:50.4370860Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/parser.py", line 561, in _report_lines_with_errors
2024-05-02T02:17:50.4371696Z     raise errors[0]  # guaranteed to have at least one exception
2024-05-02T02:17:50.4372564Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/parser.py", line 335, in parse_coverage
2024-05-02T02:17:50.4373365Z     _parse_line(raw_line, ignore_parse_errors, persistent_states),
2024-05-02T02:17:50.4374223Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/parser.py", line 678, in _parse_line
2024-05-02T02:17:50.4375014Z     tag = _parse_tag_line(line, ignore_parse_errors, persistent_states)
2024-05-02T02:17:50.4375902Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/parser.py", line 799, in _parse_tag_line
2024-05-02T02:17:50.4376628Z     NegativeHits.raise_if_not_ignored(
2024-05-02T02:17:50.4377443Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/parser.py", line 245, in raise_if_not_ignored
2024-05-02T02:17:50.4378166Z     raise NegativeHits(line)
2024-05-02T02:17:50.4379596Z gcovr.formats.gcov.parser.NegativeHits: Got negative hit value in gcov line 'branch  1 taken -8' caused by a bug in gcov tool, see https://gcc.gnu.org/bugzilla/show_bug.cgi?id=68080. Use option --gcov-ignore-parse-errors with a value of negative_hits.warn, or negative_hits.warn_once_per_file.
2024-05-02T02:17:50.4380911Z (ERROR) Error occurred while reading reports:
2024-05-02T02:17:50.4381256Z Traceback (most recent call last):
2024-05-02T02:17:50.4381925Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/__main__.py", line 362, in main
2024-05-02T02:17:50.4382593Z     covdata: CovData = gcovr_formats.read_reports(options)
2024-05-02T02:17:50.4383408Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/__init__.py", line 66, in read_reports
2024-05-02T02:17:50.4384120Z     covdata = GcovHandler(options).read_report()
2024-05-02T02:17:50.4384928Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/__init__.py", line 202, in read_report
2024-05-02T02:17:50.4385629Z     return read_report(self.options)
2024-05-02T02:17:50.4386516Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/read.py", line 79, in read_report
2024-05-02T02:17:50.4387181Z     with Workers(
2024-05-02T02:17:50.4387871Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/workers.py", line 173, in __exit__
2024-05-02T02:17:50.4388537Z     self.wait()
2024-05-02T02:17:50.4389195Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/workers.py", line 164, in wait
2024-05-02T02:17:50.4389857Z     raise self.exceptions[0][1]
2024-05-02T02:17:50.4390722Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/read.py", line 86, in read_report
2024-05-02T02:17:50.4391395Z     contexts = pool.wait()
2024-05-02T02:17:50.4392088Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/workers.py", line 164, in wait
2024-05-02T02:17:50.4392749Z     raise self.exceptions[0][1]
2024-05-02T02:17:50.4393478Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/workers.py", line 80, in worker
2024-05-02T02:17:50.4394136Z     work(*args, **kwargs)
2024-05-02T02:17:50.4394872Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/read.py", line 422, in process_datafile
2024-05-02T02:17:50.4395585Z     done = run_gcov_and_process_files(
2024-05-02T02:17:50.4396411Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/read.py", line 709, in run_gcov_and_process_files
2024-05-02T02:17:50.4397246Z     process_gcov_data(fname, abs_filename, covdata, options)
2024-05-02T02:17:50.4398184Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/read.py", line 217, in process_gcov_data
2024-05-02T02:17:50.4398976Z     coverage, source_lines = parse_coverage(
2024-05-02T02:17:50.4399853Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/parser.py", line 379, in parse_coverage
2024-05-02T02:17:50.4400610Z     _report_lines_with_errors(
2024-05-02T02:17:50.4401821Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/parser.py", line 561, in _report_lines_with_errors
2024-05-02T02:17:50.4402738Z     raise errors[0]  # guaranteed to have at least one exception
2024-05-02T02:17:50.4403696Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/parser.py", line 335, in parse_coverage
2024-05-02T02:17:50.4404562Z     _parse_line(raw_line, ignore_parse_errors, persistent_states),
2024-05-02T02:17:50.4405499Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/parser.py", line 678, in _parse_line
2024-05-02T02:17:50.4406361Z     tag = _parse_tag_line(line, ignore_parse_errors, persistent_states)
2024-05-02T02:17:50.4407328Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/parser.py", line 799, in _parse_tag_line
2024-05-02T02:17:50.4408111Z     NegativeHits.raise_if_not_ignored(
2024-05-02T02:17:50.4409008Z   File "/home/runner/.local/pipx/venvs/gcovr/lib/python3.10/site-packages/gcovr/formats/gcov/parser.py", line 245, in raise_if_not_ignored
2024-05-02T02:17:50.4409793Z     raise NegativeHits(line)
2024-05-02T02:17:50.4411333Z gcovr.formats.gcov.parser.NegativeHits: Got negative hit value in gcov line 'branch  1 taken -8' caused by a bug in gcov tool, see https://gcc.gnu.org/bugzilla/show_bug.cgi?id=68080. Use option --gcov-ignore-parse-errors with a value of negative_hits.warn, or negative_hits.warn_once_per_file.
2024-05-02T02:17:50.4412631Z 
2024-05-02T02:17:50.4413744Z Got negative hit value in gcov line 'branch  1 taken -8' caused by a bug in gcov tool, see https://gcc.gnu.org/bugzilla/show_bug.cgi?id=68080. Use option --gcov-ignore-parse-errors with a value of negative_hits.warn, or negative_hits.warn_once_per_file.
2024-05-02T02:17:50.4425801Z ##[endgroup]
2024-05-02T02:17:50.4449943Z ##[error]Error: Failed to generate code coverage report: coverage is under %
2024-05-02T02:17:50.5764205Z Received 10843072 of 10843072 (100.0%), 10.3 MBs/sec
2024-05-02T02:17:50.5944183Z Post job cleanup.
2024-05-02T02:17:50.6891809Z [command]/usr/bin/git version
2024-05-02T02:17:50.6941207Z git version 2.43.2
2024-05-02T02:17:50.6980594Z Temporarily overriding HOME='/home/runner/work/_temp/57613229-2ef0-4ecb-b778-c7c784752c34' before making global git config changes
2024-05-02T02:17:50.6982273Z Adding repository directory to the temporary git global config as a safe directory
2024-05-02T02:17:50.6986928Z [command]/usr/bin/git config --global --add safe.directory /home/runner/work/gz-sim-yarp-plugins/gz-sim-yarp-plugins
2024-05-02T02:17:50.7026442Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2024-05-02T02:17:50.7064195Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
2024-05-02T02:17:50.7295986Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2024-05-02T02:17:50.7324726Z http.https://github.com/.extraheader
2024-05-02T02:17:50.7334102Z [command]/usr/bin/git config --local --unset-all http.https://github.com/.extraheader
2024-05-02T02:17:50.7369247Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :"
2024-05-02T02:17:50.7863574Z Cleaning up orphan processes
~~~

Given the limited help of the code coverage step, personally I am in favor of disabling it.

--- a/.github/workflows/apt.yml
+++ b/.github/workflows/apt.yml
@@ -8,7 +8,7 @@ on:
   pull_request:
   schedule:
   # * is a special character in YAML so you have to quote this string
-  # Execute a "nightly" build at 2 AM UTC 
+  # Execute a "nightly" build at 2 AM UTC
   - cron:  '0 2 * * *'
 
 jobs:
@@ -26,11 +26,11 @@ jobs:
 
     steps:
     - uses: actions/checkout@v2
-        
+
     # Print environment variables to simplify development and debugging
     - name: Environment Variables
       run: env
-        
+
     # ============
     # DEPENDENCIES
     # ============
@@ -63,7 +63,7 @@ jobs:
         mkdir -p build
         cd build
         cmake -GNinja -DCMAKE_INSTALL_PREFIX=${GITHUB_WORKSPACE}/install/deps ..
-        cmake --build . --config ${{ matrix.build_type }} --target install 
+        cmake --build . --config ${{ matrix.build_type }} --target install
         # YARP
         cd ${GITHUB_WORKSPACE}
         git clone https://github.com/robotology/yarp
@@ -73,12 +73,12 @@ jobs:
         cd build
         cmake -GNinja -DCMAKE_PREFIX_PATH=${GITHUB_WORKSPACE}/install/deps -DCMAKE_INSTALL_PREFIX=${GITHUB_WORKSPACE}/install/deps ..
         cmake --build . --config ${{ matrix.build_type }} --target install
-        
+
     - name: Configure Environment variables for Source-based Dependencies
       run: |
         # Configure environment
         echo "YARP_DATA_DIRS=${GITHUB_WORKSPACE}/install/deps/share/yarp" >> $GITHUB_ENV
- 
+
     # ===================
     # CMAKE-BASED PROJECT
     # ===================
@@ -87,14 +87,14 @@ jobs:
       shell: bash
       run: |
         mkdir -p build
-        cd build    
+        cd build
         cmake -GNinja -DCMAKE_PREFIX_PATH=${GITHUB_WORKSPACE}/install/deps -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} \
               -DCMAKE_CXX_FLAGS_DEBUG="--coverage -g -O0" -DCMAKE_INSTALL_PREFIX=${GITHUB_WORKSPACE}/install -DBUILD_TESTING:BOOL=ON ..
 
     - name: Build
       run: |
         cd build
-        cmake --build . --config ${{ matrix.build_type }} 
+        cmake --build . --config ${{ matrix.build_type }}
 
     - name: Test (Release)
       if: contains(matrix.build_type, 'Release')
@@ -108,24 +108,8 @@ jobs:
       run: |
         cd build
         # So high until-pass as a mitigation for https://github.com/robotology/gz-sim-yarp-plugins/issues/75
-        ctest -T Test -T Coverage --repeat until-pass:20 --output-on-failure -C ${{ matrix.build_type }} . 
-      
-    - name: Generate a code coverage report
-      if: contains(matrix.build_type, 'Debug')
-      uses: threeal/gcovr-action@main
-      with:
-        excludes: |
-          build/*
-          install/*
-        # fail-under-line: 80
-    
-    - name: Upload coverage reports to Codecov
-      if: contains(matrix.build_type, 'Debug')
-      uses: codecov/codecov-action@v4
-      env:
-        CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
+        ctest -T Test -T Coverage --repeat until-pass:20 --output-on-failure -C ${{ matrix.build_type }} .
 
-    
     - name: Install
       run: |
         cd build

